hashicorp/boundary-ui
# feat: 🎸 Setup initial desktop E2E tests

✅ Closes: https://hashicorp.atlassian.net/browse/ICU-14061

# Description
Sets up the initial end to end tests for desktop client. I didn't use a helper class but instead integrated them as fixtures in combination with [page object models](https://playwright.dev/docs/pom).

These fixtures get loaded as part of each test and automatically setup and tear down the things needed for the electron app to be tested as well as provide helpers for common actions.

## Screenshots (if appropriate)
Forced Failure:
<img width="1017" alt="image" src="https://github.com/user-attachments/assets/18a3d8d0-f12d-4d0c-977d-d684882130f2">

<img width="1013" alt="image" src="https://github.com/user-attachments/assets/a0994f78-0712-4e3c-b7ce-27f8fe28b6c6">

<img width="1392" alt="image" src="https://github.com/user-attachments/assets/4732e18d-8e33-4ce2-a848-0df16852f1ce">

## How to Test
Run `npx playwright test --reporter=html` to get an html report. Force an error by adding something like `expect(1).toBe(2)` to a test to see an example of a trace and screenshot that gets captured.

Note that the trace is a bit buggy though the html viewer (still works, just images/css not loaded correctly for the snapshots though screenshots look fine). You can see it rendered properly by taking the trace in `tests/e2e/artifacts` and passing it to `npx playwright show-trace`

## Checklist
